apcd-cms/src/client/src/components/Forms/Registrations/RegistrationForm.tsx

```diff
@@ -181,48 +154,39 @@ const initialValues: FormValues = {
       contact_phone: '',
       contact_email: '',
       contact_notifications: false,
+      contact_id: -1,
     },
   ],
 };
 
-export const RegistrationForm: React.FC = () => {
-  const { data, isLoading, isError } = useRegFormData({});
-  const [errorMessage, setErrorMessage] = useState('');
-  const [isSuccess, setIsSuccess] = useState<boolean>(false);
-  const [regId, setRegId] = useState(null);
+export const RegistrationForm: React.FC<{
+  isEdit?: boolean;
+  inputValues?: RegistrationFormValues;
+  isModal?: boolean;
+  onSuccessCallback?: () => void;
+}> = ({
+  isEdit = false,
+  inputValues,
+  isModal = false,
+  onSuccessCallback = () => {},
+}) => {
+  const [searchParams] = useSearchParams();
+  const { data, isLoading, isError } = useRegFormData(
+    searchParams.get('reg_id')
+  );
+  const {
+    mutate: submitForm,
+    isLoading: registrationSubmissionPending,
+    isSuccess,
+    data: registrationResponse,
+    error: registrationError,
+  } = usePostRegistration();
 
-  const handleSubmit = async (
```

Comment:
All of this is moved to hook to enable easy `invalidation` of other existing query related data.